src/Fantomas/AstTransformer.fs

```diff
@@ -301,13 +301,11 @@ module private Ast =
                         mkNode SynExpr_Lazy range :: nodes
                         |> finalContinuation)
             | SynExpr.Sequential (_, _, expr1, expr2, _) ->
-                let continuations : ((TriviaNodeAssigner list -> TriviaNodeAssigner list) -> TriviaNodeAssigner list) list =
-                    [ visit expr1; visit expr2 ]
-
-                let finalContinuation (nodes: TriviaNodeAssigner list list) : TriviaNodeAssigner list =
-                    (List.collect id nodes) |> finalContinuation
-
-                Continuation.sequence continuations finalContinuation
+                visit
+                    expr2
+                    (fun nodes1 ->
+                        visit expr1 (fun nodes2 -> nodes1 @ nodes2)
+                        |> finalContinuation)
```

Comment:
This one was a bit counter-intuitive, I thought I would need to call `finalContinuation` inside the second lambda.
Had to call it in the first lambda to not blow the stack.

`expr2` in the unit test will again be a `SynExpr.Sequentials`.

Does all of this make some sense?

Reply:
I am deeply confused. I would expect to call `finalContinuation` inside the second lambda, yes. Does the situation improve in Release mode (rather than Debug mode)?

Reply:
Release mode works just fine, thanks for the pointer.